### PR TITLE
Refactor API to be "production ready"

### PR DESCRIPTION
**Pull request by Kondax**
I was going to write a full explanation however this has taken a lot of time already, so here are some notes from when I was working on it:

There's some fundamental flaws with tron-web, namely:
- Lack of naming conventions. Function names seem to be almost random, this might be a translation issue.
- Lack of the "accepted" names. Super Representative / Witness is used freely, whereas the API should only use one or the other
- Documentation is poorly written. This also most likely stems from poor translation.
- Functions aren't namespaced. Ideally any "create transaction" functions would be under a single namespace, same with querying blockchain data etc.
- A lot of functions return objects where they could simply return a single parameter. `getnextmaintenancetime` could return the time in milliseconds whereas it returns `{ num: timestamp }`

Removed:
- `easyTransferByPrivate`: Users can sign transaction locally instead. This is insecure and there's no reason to do this over the net.
- `generateAddressOnLine`: This is insecure and can be done offline. The offline function was included. Why was this there in the first place? This should probably be removed from the node API.

Regarding `validateAddress`: Why is this online? This doesn't make sense.

I've removed the try/catch statements. The errors should be passed to the developer so that they can take appropriate action.

I've essentially re-written a lot of functions and the jsdocs for them to be better suited for production development. I've also re-written most of the library so that other contributors can understand the code.

Changed:
- `login(pk)` -> `privateKeyToAddress(privateKey)`
- `blockNumber()` -> `latestBlockNumber()`
- `getBlock(hashStringOrblockNumber)` -> `getBlock(blockIdentifier)`
- `getBlockTransactionCount(hashStringOrBlockNumber)` -> `getBlockTransactionCount(blockIdentifier)`
- `getTransaction(id)` -> `getTransaction(transactionID)`
- `sendTransaction(from, to, amount, privateKey)` -> `sendTransaction(from, to, amount, privateKey = this.defaultPrivateKey)`
- `createTransaction(to_address, owner_address, amount)` -> `createTransaction(from, to, amount)`
- `signTransaction(transaction, privateKey, terminal)` -> `signTransaction(transaction, privateKey)`
- `sendRawTransaction(signTransaction)` -> `sendRawTransaction(signedTransaction)`
- `updateAccount(account_name, owner_address)` -> `changeAccountName(address, newName)`
-  `voteWitnessAccount(owner_address, votes)` -> `voteWitnessAccount(address, votes)`
- `createToken(oToken)` -> `createToken(token)`
- `createAccount(owner_address, account_address)` -> `registerAccount(address, newAccountAddress)`
- `createWitness(owner_address, url)` -> `applyForSuperRepresentative(address, url)`
- `transferAsset(options)` -> `createSendAssetTransaction(from, to, assetID, amount)`
- `easytransfer(passPhrase, toAddress, amount)` -> `sendTransactionByPassword(to, amount, password)` **I wanted to remove this as it's not secure, but there's no equivalent locally at the moment**
- `createAddress(password)` -> `createAddressWithPassword(password)` **Same as above**
- `participateAssetIssue(options)` -> `createPurchaseAssetTransaction(tokenIssuer, address, amount, assetID)`
- `freezeBalance(owner_address,frozen_balance,frozen_duration,resource='BANDWIDTH')` -> `createFreezeBalanceTransaction(address, amount, duration = 3)`
- `unfreezeBalance(owner_address)` -> `createUnfreezeBalanceTransaction(address)`
- `unfreezeAsset(owner_address)` -> `createUnfreezeAssetTransaction(address)`
- `withdrawBalance(owner_address)` -> `createWithdrawBlockRewardTransaction(address)`
- `updateAsset(options)` -> `createUpdateAssetTransaction(address, description, url, bandwidthLimit = 0, freeBandwidthLimit = 0)`
- `getAssetIssueByAccount(address)` -> `getTokenFromAddress(address)`
- `getAccountNet(address)` -> `getAccountBandwidth(address)`
- `getAssetIssueByName(value)` -> `getAssetIssueFromName(name)`
- `getBlockByLimitNext(startNum,endNum)` -> `getBlocksInRange(startBlock, endBlock)`
- `getBlockByLatestNum(num)` -> `getLatestBlocks(limit = 1)`
- `listWitNesses()` -> `listSuperRepresentatives()`
- `getAssetIssueList()` -> `listAssets()`
- `getPaginateDassetIssueList(offset, limit)` -> `listAssetsPaginated(limit, offset = 0)`
- `getNextMainteNanceTime()` -> `timeUntilNextVoteCycle()`
- `deployContract(options)` -> `createDeployContractTransaction(abi, bytecode, feeLimit, address, callValue = 0, bandwidthLimit = 0)`
- `triggerSmartContract(options)` -> `createTriggerContractTransaction(contractAddress, functionSelector, callValue, feeLimit, address, parameters = [])`
- `getEventResult({contractAddress,eventName,blockNum,transactionId})` -> `getEventResult(contractAddress = false, eventName = false, blockNumber = false, transactionID = false)`

Added:
- `getBlockByHash(blockHash)`
- `getBlockByNumber(blockNumber)`

Removed (for security reasons):
- `easyTransferByPrivate()`
- `generateAddressOnLine()`

Overall the API should be a lot easier to work with and not so daunting for newcomers. API functions have been renamed to give a better description of their functionality along with adequate jsdocs.

This has been discussed privately with @kookiekrak 

This is a proposal and I cannot guarantee that there are 0 errors associated with my changes. I have done the best I can to verify that my changes are still fully working.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "tronweb",
   "version": "1.1.1",
-  "description": "A jssdk that encapsulates the TRON HTTP request",
+  "description": "JavaScript SDK that encapsulates the Tron Node HTTP API",
```
Comment (kookiekrak) [inline package.json]: can you use TRON (marketing style guides)

File: src/index.js [hunk 2]
```diff
@@ -243,500 +269,513 @@ class TronWeb {
...
 
-    toBigNumber(str){
-        return BigNumber(str)
+    toBigNumber(str) {
+        return BigNumber(str);
     }
-    sendTransactionByWallet(options,callback){
+
+    // Leaving this untouched as I'm unawaure of the wallet API you're using
+    sendTransactionByWallet(options, callback) {
```
Comment (kookiekrak) [inline src/index.js]: yeah...

maybe revise this to just note that its a special integration for oTronWallet (not sure what that is)

File: package.json (hunk 1 above)
Comment (Kondax) [reply package.json]: Of course. Let me add now.

File: src/index.js (hunk 2 above)
Comment (Kondax) [reply src/index.js]: Sure thing. As far as I'm aware it's using https://github.com/1119756420/Treasure for internal testing.

**Comment by Kondax** [comment]
Also, do we want to ship the built files? This would allow developers to include the project (with npm or through cdn) to the browser without having to build first.

**Comment by kookiekrak** [comment]
@Kondax yeah absolutely

**Pull request merged**